config/pmd.xml

```diff
@@ -7,320 +7,609 @@
-    <exclude name="ModifiedCyclomaticComplexity"/>
-    <!-- we are using CyclomaticComplexity -->
-    <exclude name="StdCyclomaticComplexity"/>
+
+  <!-- We use a custom rule instead of ShortVariable to control length and skip @Override methods -->
+  <rule name="CheckstyleCustomShortVariable"
+        message="Avoid variables with short names that shorter than 2 symbols: {0}"
+        language="java"
+        class="net.sourceforge.pmd.lang.rule.XPathRule"
+        externalInfoUrl="">
+  <description>
+     Fields, local variables, or parameter names that are very short are not helpful to the reader.
+  </description>
+  <priority>3</priority>
+  <properties>
+     <property name="xpath">
+         <value>
+         <![CDATA[
+//VariableDeclaratorId[string-length(@Image) < 2]
+[not(ancestor::ForInit)]
+[not(../../VariableDeclarator and ../../../LocalVariableDeclaration and ../../../../ForStatement)]
+[not((ancestor::FormalParameter) and (ancestor::TryStatement))]
+[not(ancestor::ClassOrInterfaceDeclaration[//MarkerAnnotation/Name[pmd-java:typeof(@Image, 'java.lang.Override', 'Override')]])]
+         ]]>
+         </value>
+     </property>
+  </properties>
   </rule>
 
-  <rule ref="rulesets/java/codesize.xml/CyclomaticComplexity">
-    <properties>
-      <property name="showClassesComplexity" value="false"/>
-      <property name="reportLevel" value="11"/>
-      <!-- DeclarationOrder - 'visitToken' has just big SWITCH block which contains IF blocks.
-           If we split the block to several methods it will damage readability.
-           validateCli is not reasonable to split as encapsulation of logic will be damaged
-           getDetails - huge Switch, it has to be monolithic
-           JavadocMethodCheck, JavadocStyleCheck, JavadocUtils.getJavadocTags() - deprecated
-           FinalLocalVariableCheck.visitToken() - it is just big switch , not could be done
-           GenericWhitespaceCheck.processSingleGeneric() - it contains a lot of similar to each other and very simple condition
-           VariableDeclarationUsageDistanceCheck, CustomImportOrderCheck - it is not OK to have such a complicated logic need to be refactored.
-      -->
-      <property name="violationSuppressXPath" value="//DeclarationOrder[@Name='visitToken'] and //MethodDeclaration[@Name='validateCli' and ../../..[@Image='Main']] | //MethodDeclaration[@Name='processSingleGeneric' and ../../..[@Image='GenericWhitespaceCheck']] | //MethodDeclaration[@Name='visitToken' and ../../..[@Image='FinalLocalVariableCheck']] | //MethodDeclaration[@Name='getJavadocTags' and ../../..[@Image='JavadocUtils']] | //MethodDeclaration[@Name='getDetails' and ../../..[@Image='RightCurlyCheck']] | //ClassOrInterfaceDeclaration[@Image='JavadocMethodCheck' or @Image='JavadocStyleCheck' or @Image='VariableDeclarationUsageDistanceCheck' or @Image='CustomImportOrderCheck']"/>
-    </properties>
+  <!-- Best practices. The list of rules in this category:
```

Comment:
please remove such blocks from all parts of config.
Such list is not maintainable. On next pmd release they will introduce new rule and if it does not make any violation, nobody will notice it, so list will become out-dated.

Nobody will read pmd release notes attentively, to keep this list up-to-date.